guide/high-level-da-workflows.rst

```diff
@@ -0,0 +1,155 @@
+“Assimilate” portion of the above diagram and/or the “diagnostics” as follows:
+
+|complex_model_workflow|
+
+.. |complex_model_workflow| image:: images/complex_model_workflow.jpg
+   :width: 100%
+
+The single time-step workflow for an OSE experiment within a single step is
+slightly simpler than the OSSE equivalent as DART handles less of the process.
+Like the OSSE case, the namelist and preprocessed observation source files are
+input to all other DART programs. In the OSE case, however, the user must
+provide an **obs converter** that will output a ``obs_seq.out`` file. There are
+many DART utilities to make this process easier, but for the OSE case the
+``obs_seq.out`` file is ultimately the user’s responsibility (to avoid
+duplicating effort, see the list of existing observation types in :doc:`Important
+capabilities of DART <important-capabilities-dart>`). Here, the option to run with one
+file for each ensemble member is demonstrated. There are *k* ensemble members
+used as input to *filter*, which also outputs *k* members for the prior and
+posterior. The ``obs_seq.final`` and ``obs_diag_output.nc`` are used in the same
+way as in the OSSE case. The names of the input files and output files can be
+controlled by the user through the ``filter_input_list.txt`` and
+``filter_output_list.txt`` files, which can contain the user-specified list of
+the ensemble input or output files, respectively.
+
+Another view of the stages of filter is shown in the following diagram:
+
+|diag_cycle|
+
+.. |diag_cycle| image:: images/diag_cycle.png
+   :width: 100%
+
+As shown here, an ensemble forecast is stored in ``forecast.nc`` , to which
+prior inflation can be applied and stored in ``preassim.nc``. Once assimilation
+is applied, the output can be stored in ``postassim.nc``, and finally if
+posterior inflation is applied, the final analysis can be written in
+``analysis.nc`` . The model forecast will start from the analysis to advance the
+model in order to start the cycle over again.
+
+.. note::
+
+   The “forecast” will be the same as the “preassim” if prior inflation is not
+   used, and the “postassim” will be the same as the “analysis” if posterior
+   inflation is not used. The *stages_to_write* variable in the “&filter_nml”
+   section of the ``input.nml`` namelist controls which stages are output to
+   file. For a multi-file case, the potential *stages_to_write* are “input,
+   forecast, preassim, postassim, analysis, output” while for a single file the
+   same stages are available with the exception of “input.”
+
+.. note::
+    
+   In the above cycling diagram, there will actually be one file per member,
+   which is not shown here in order to simplify the process.
+
+.. important:: 
+
+   The decision to store ensemble members as separate files and whether to run
+   an OSSE or OSE are independent. An OSSE can be run with multiple files and
+   an OSE can be run with all ensemble members stored in a single file.
```

Comment:
Although there are restrictions on single-file experiments - non-subroutine-callable models (I believe) MUST use multiple files for their ensemble members.

Reply:
i think as long as you only do a single assimilation window inside filter there are no restrictions.  the restriction is that if you have a non-subroutine callable model you cannot do more than a single assimilation in filter.   if you do have subroutine callable model you have to use the single-file (combination file - all copies in one netcdf file of our formatting) to cycle.  in practice, if your state is large you probably want to use multiple files, but that's a choice not a requirement.